src/Firebase/AppCheck/AppCheckTokenVerifier.php

```diff
@@ -0,0 +1,96 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Kreait\Firebase\AppCheck;
+
+use Firebase\JWT\CachedKeySet;
+use Firebase\JWT\JWT;
+use Kreait\Firebase\Exception\AppCheck\FailedToVerifyAppCheckToken;
+use Kreait\Firebase\Exception\AppCheck\InvalidAppCheckToken;
+use LogicException;
+use UnexpectedValueException;
+
+use function in_array;
+use function is_string;
+use function str_starts_with;
+
+/**
+ * @internal
+ *
+ * @phpstan-import-type DecodedAppCheckTokenShape from DecodedAppCheckToken
+ */
+class AppCheckTokenVerifier
+{
+    private const APP_CHECK_ISSUER_PREFIX = 'https://firebaseappcheck.googleapis.com/';
+    private string $projectId;
+    private CachedKeySet $keySet;
+
+    /**
+     * @param non-empty-string $projectId
+     */
+    public function __construct(string $projectId, CachedKeySet $keySet)
+    {
+        $this->projectId = $projectId;
+        $this->keySet = $keySet;
+    }
+
+    /**
+     * Verfies the format and signature of a Firebase App Check token.
```

Comment:
```suggestion
     * Verifies the format and signature of a Firebase App Check token.
```

When I start to get nitpicky, it's usually a good sign 😂